kubernetes/resource_kubernetes_cron_job.go

```diff
@@ -0,0 +1,211 @@
+package kubernetes
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgApi "k8s.io/apimachinery/pkg/types"
+	kubernetes "k8s.io/client-go/kubernetes"
+	batchv2 "k8s.io/client-go/pkg/apis/batch/v2alpha1"
+)
+
+func resourceKubernetesCronJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKubernetesCronJobCreate,
+		Read:   resourceKubernetesCronJobRead,
+		Update: resourceKubernetesCronJobUpdate,
+		Delete: resourceKubernetesCronJobDelete,
+		Exists: resourceKubernetesCronJobExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"metadata": namespacedMetadataSchema("cronjob", true),
+			"spec": {
+				Type:        schema.TypeList,
+				Description: "Spec of the cron job owned by the cluster",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: cronJobSpecFields(),
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesCronJobCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	spec, err := expandCronJobSpec(d.Get("spec").([]interface{}))
+	if err != nil {
+		return err
+	}
+	spec.JobTemplate.ObjectMeta.Annotations = metadata.Annotations
```

Comment:
FYI - I'm going to update the JobSpec to better match the native Kubernetes API, so it will be possible to specify the Metdata within the JobTemplateSpec. This means lines like this will be unnecessary.